scripts/update-wallet-env.js

```diff
@@ -0,0 +1,62 @@
+const fs = require('node:fs')
+const path = require('node:path')
+
+async function rpcRequest(method, params = []) {
+  const result = await fetch(
+    'https://mainnet.ckb.dev/rpc',
+    {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({
+        id: 0,
+        jsonrpc: '2.0',
+        method,
+        params,
+      }),
+      timeout: 10000,
+    }
+  )
+  return result.json()
+}
+
+const ESTIMATE_BLOCK_COUNT_PER_DAY = 8_000
+const envFilePath = path.resolve(__dirname, '../packages/neuron-wallet/.env')
+const validTargetReg = /(CKB_NODE_ASSUME_VALID_TARGET=)[\S]*/
+const ckbNodeDataSizeReg = /(CKB_NODE_DATA_SIZE=)[\S]*/
+
+async function getValidTarget() {
+  const tipBlockNumber = (await rpcRequest('get_tip_block_number')).result
+  const validTargetBlockNumber = `0x${(BigInt(tipBlockNumber) - BigInt(ESTIMATE_BLOCK_COUNT_PER_DAY)).toString(16)}`
+  return (await rpcRequest('get_block_hash', [validTargetBlockNumber])).result
+}
+
+async function getCKBNodeSize() {
+  const originEnvContent = fs.readFileSync(envFilePath).toString('utf-8')
+  const validTarget = originEnvContent.match(validTargetReg)[0].split('=')
+  if (validTarget[1]) {
+    const blockInfo = await rpcRequest('get_block', [validTarget[1].replaceAll("'", '')])
+    const lastBlockNumber = blockInfo.result.header.number
+    const tipBlockNumber = (await rpcRequest('get_tip_block_number')).result
+    const lastCkbNodeDataSize = originEnvContent.match(ckbNodeDataSizeReg)[0].split('=')[1]
+    return +lastCkbNodeDataSize + 4 * (BigInt(tipBlockNumber) - BigInt(lastBlockNumber)).toString()
```

Comment:
Where is the rule defined, I thought it was updated on the project page and fetched from there.

Ref: https://github.com/Magickbase/neuron-public-issues/issues/255#issuecomment-1789975492

---

Besides, it's better to add a link to the rule document for further review.

Reply:
This is just an estimated value. We can manually adjust it. Where can we get the actual data size?

Reply:
Mixing the `bigint`, `number`, and `string` in arithmetic is not a good practice. The `Number.MAX_SAFE_INTEGER` is OK to calculate the size

```js
(Number(tipBlockNumber) - Number(lastBlockNumber))
```

Reply:
> This is just an estimated value. We can manually adjust it. Where can we get the actual data size?

What I expected before was
1. We have a running node
2. The running node submits the size periodically to the neuron project page, displayed on the download page
3. Neuron repo fetches the size from the project page and updates the repo periodically.

Reply:
Is the project online or in the plan?

Reply:
`Updating node size` on the project page should be included in this task, but assigned to other members, one to expose the API to fetch node size from our node, one to add a CI to update the page

Reply:
That's cool, after the API is ready, I will reopen the PR. Currently, I will set it to draft.